sources/web/datalab/polymer/modules/gapi-manager/gapi-manager.ts

```diff
@@ -13,31 +13,47 @@
  */
 
 /// <reference path="../../../../../../third_party/externs/ts/gapi/bigquery.d.ts" />
-/// <reference path="../../../../../../third_party/externs/ts/gapi/gapi.d.ts" />
-/// <reference path="../../../common.d.ts" />
 
 class MissingClientIdError extends Error {
   message = 'No oauth2ClientId found in user or config settings';
 }
 
+enum GapiScopes {
+  BIGQUERY,
+  DRIVE,
+  GCS,
+}
+
+const initialScopeString = 'profile email';
+
 /**
  * This file contains a collection of functions that interact with gapi.
  */
 class GapiManager {
 
-  public static DISCOVERYDOCS = [
-    'https://www.googleapis.com/discovery/v1/apis/drive/v3/rest',
-    'https://www.googleapis.com/discovery/v1/apis/bigquery/v2/rest'
-  ];
-  public static SCOPES =
-      'https://www.googleapis.com/auth/drive.metadata.readonly ' +
-      'https://www.googleapis.com/auth/devstorage.full_control ' +
-      'https://www.googleapis.com/auth/bigquery';
-
   private static _clientId = '';   // Gets set by _loadClientId()
   private static _loadPromise: Promise<void>;
 
   /**
+   * Request a new scope to be granted.
+   */
+  public static async grantScope(scope: GapiScopes): Promise<any> {
+    await this.loadGapi();
```

Comment:
Pages that require a specific scope should call the `grantScope` method, which will ask the user to sign in with this scope if they're signed out. The basic profile requested in the separate Sign In operation does not require the user to approve anything.